debian/logitechmediaserver.service

```diff
@@ -0,0 +1,174 @@
+#=======================================================================
+# Unit section
+#=======================================================================
+
+[Unit]
+
+Description=Logitech Media Server
+
+# Ensures LMS is shutdown before network. And that the network service
+# has at least been started.
+
+After=network.target
+
+# Make LMS wait for network-online.target (if it gets started).
+# See notes below.
+
+After=network-online.target
+
+#-----------------------------------------------------------------------
+# Some notes on networking targets:
+#-----------------------------------------------------------------------
+
+# We leave it to the user to specify 'Wants=network-online.target' if he
+# chooses, because it may not suit all users. But we specify
+# 'After=network-online.target' because the SysV init script has
+# essentially behaved that way by virtue of the 'Required-Start: $all'
+# stanza included within it.
+# This is also the approach taken in Debian's & Ubuntu's implementation
+# of 'rc.local.service'.
+#
+# On systemd >= 217, user might execute:
+#   'systemctl add-wants logitechmediaserver.service network-online.target'
+#
+# The effects of the 'After' and 'Wants' 'network-online' stanzas are
+# very system dependent. They may or may not ensure that LMS starts up
+# after network connections have been established, and they may or may
+# not cause a delay in LMS start up.
+#
+# 'After=network-online.target' schedules LMS to start after the network
+# is fully up, whatever that may mean. But it is only effective if the
+# 'network-online.target' is active and, even then, it is only effective
+# if the user's system has been, or can be, configured to wait
+# appropriately for network connections to be established before
+# reaching this target.
+#
+# 'Wants=network-online.target' would activate (pull in) that target.
+#
+
+#=======================================================================
+# Service section
+#=======================================================================
+
+[Service]
+
+#-----------------------------------------------------------------------
+# Set run time variables, essentially replicating the SysV init script.
+#-----------------------------------------------------------------------
+
+# Use of '/etc/default/logitechmediaserver' to set run time variables
+# is supported, for consistency with the init script.
+# But setting 'SLIMUSER' is not supported. This unit file will always
+# start LMS as user 'squeezeboxserver'.
+# The user may override that, the methods available depend on the
+# installed version of systemd.
+
+# Note one consequence of this approach: LMS can now see environment
+# variables that correspond to the defined run time variables, should it
+# choose to look (it doesn't at present). This behaviour is an artefact
```

Comment:
What do you mean by "defined run time variables" ? 
I'd have thought that systemd mean less leakeage of environment between invoking environment and runtime.

Reply:
I mean the shell variables that are defined in the init script and defaults file. These are used to build the LMS command line. Perhaps I might have said "defined run time parameters". Or something else. I'll try to find some better words.

A systemd unit file does not provide for parameter substitution (no "local variables"), so I am using environment variables to achieve the same trick, i.e. read the defaults file to set environment variables, which we _can_ then use. But that, of necessity, leads to a rather full environment. I believe that some other projects may have adopted the same approach.

This is all predicated on the notion that we really do wish to support use of the existing defaults file. I suspect that its main role is in defining _SLIMOPTIONS_.

Another way of doing it would be to abandon the defaults file altogether and teach users how to edit their unit files, possibly after explaining to them why the upgrade didn't "just work"... :(